sky/provision/kubernetes/instance.py

```diff
@@ -92,6 +92,49 @@ def _raise_pod_scheduling_errors(namespace, new_nodes):
     are recorded as events. This function retrieves those events and raises
     descriptive errors for better debugging and user feedback.
     """
+
+    def _formatted_resource_requirements(pod):
+        # Returns a formatted string of resource requirements for a pod.
+        resource_requirements = {}
+        for container in pod.spec.containers:
+            for resource, value in container.resources.requests.items():
+                if resource not in resource_requirements:
+                    resource_requirements[resource] = 0
+                if resource == 'memory':
+                    int_value = kubernetes_utils.parse_memory_resource(value)
+                else:
+                    int_value = kubernetes_utils.parse_cpu_or_gpu_resource(
+                        value)
+                resource_requirements[resource] += int_value
+        return ', '.join(f'{resource}={value}'
+                         for resource, value in resource_requirements.items())
+
+    def _formatted_node_selector(pod) -> Optional[str]:
+        # Returns a formatted string of node selectors for a pod.
+        node_selectors = []
+        if pod.spec.node_selector is None:
+            return None
+        for label_key, label_value in pod.spec.node_selector.items():
+            node_selectors.append(f'{label_key}={label_value}')
+        return ', '.join(node_selectors)
+
+    def lack_resource_msg(resource, pod, extra_msg=None, details=None) -> str:
+        resource_requirements = _formatted_resource_requirements(pod)
+        node_selectors = _formatted_node_selector(pod)
+        node_selector_str = f' and labels ({node_selectors})' if (
+            node_selectors) else ''
+        msg = (
+            f'Insufficient {resource} capacity on the cluster. '
+            f'Required resources ({resource_requirements}){node_selector_str} '
+            'were not found in a single node. Other SkyPilot tasks or pods may '
+            'be using resources. Check resource usage by running '
+            '`kubectl describe nodes`.')
+        if extra_msg:
+            msg += f' {extra_msg}'
+        if details:
+            msg += f'\nFull error: {event_message}'
```

Comment:
can we replace `event_message` with `details`? It'll reference a value outside of the function scope otherwise. It looks like based on the current usage `event_message` is always being passed as kwarg to `details` in any case.

Reply:
Good catch, thanks!